sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/tests/Azure.Monitor.OpenTelemetry.Exporter.Tests/TagsTests.cs

```diff
@@ -344,6 +345,40 @@ public void ActivityTagsProcessor_CategorizeTags_ExtractsAuthUserId(ActivityKind
             Assert.Equal("TestUser", activityTagsProcessor.EndUserId);
         }
 
+        [Theory]
+        [InlineData("en-US")] // French culture
+        [InlineData("de-DE")] // German culture
+        public void TagObjects_TestCulture(string cultureName)
+        {
+            var activityTagsProcessor = new ActivityTagsProcessor();
+            var originalCulture = Thread.CurrentThread.CurrentCulture;
+            var cultureInfo = new CultureInfo(cultureName);
+            Thread.CurrentThread.CurrentCulture = cultureInfo;
+
+            var doubleArray = new double[] { 1.1, 2.2, 3.3 };
+            var doubleValue = 123.45;
+
+            IEnumerable<KeyValuePair<string, object?>> tagObjects = new Dictionary<string, object?>
+            {
+                ["doubleArray"] = doubleArray,
+                ["double"] = doubleValue,
+            };
+
+            using var activity = CreateTestActivity(tagObjects);
+            activityTagsProcessor.CategorizeTags(activity);
+
+            var result = doubleArray.ToCommaDelimitedString(cultureInfo);
+
+            Assert.NotEqual("1.1,2.2,3.3", result);
+            Assert.NotEqual("123.45", doubleValue.ToString(cultureInfo));
```

Comment:
These lines confused me.

- Is it `NotEqual` here because French and German use the comma instead of the decimal?
If so, I think this would be easier to understand if we used `Equal` here.

- nit: the `doubleArray` is stored as a `var result`, but the `doubleValue` is asserted inline.
I would recommend moving the doubleArary to also be inline. 

- nit: consider adding a comment to these two groups of Asserts. Something like "Asserting Culture Behavior" and "Asserting CultureInvariant Behavior"

Reply:
> Is it NotEqual here because French and German use the comma instead of the decimal?
If so, I think this would be easier to understand if we used Equal here.

Both of these culture could be using comma. There could be other culture which may do it differently. So apt thing to do here is not equal. If not need to move away from theory and use Fact.